modules/infra/control/loopback.c

```diff
@@ -134,6 +134,7 @@ static void iface_loopback_poll(evutil_socket_t, short reason, void *ev_iface) {
 
 	// packet sent from linux tun iface, no need to compute checksum;
 	mbuf->ol_flags = RTE_MBUF_F_RX_IP_CKSUM_GOOD;
+	mbuf->packet_type = data[0] == 6 ? RTE_PTYPE_L3_IPV6 : RTE_PTYPE_L3_IPV4;
```

Comment:
Ugh, ok.
I guess the previous patch relies on this.

This must be documented, or a different mechanism is needed.

Reply:
I wished to avoid setting unnecessary data if the packet type was already defined by the drivers.
The other option is to add a `family` field in the buffer metadata for both `ip_local_mbuf_data` and `ip6_local_mbuf_data` to define which kind of header we should fill in.
I found it uglier as the 2 priv data types are defined in different modules.

```
GR_MBUF_PRIV_DATA_TYPE(ip_local_mbuf_data, {
+     uint8_t family;
        ip4_addr_t src;
        ip4_addr_t dst;
        uint16_t len;
        uint16_t vrf_id;
        uint8_t proto;
        uint8_t ttl;
});


GR_MBUF_PRIV_DATA_TYPE(ip6_local_mbuf_data, {
+     uint8_t family;
        struct rte_ipv6_addr src;
        struct rte_ipv6_addr dst;
        uint16_t len;
        uint8_t hop_limit;
        uint8_t proto;
});
```

@rjarry Any opinion about that ? 
Should we define instead a common header for both local_mbuf_data instead ?
```
GR_MBUF_PRIV_DATA_TYPE(ip46_local_mbuf_data, {
        uint8_t family;
        union {
            struct rte_ipv6_addr v6;
            ip4_addr_t v4;  
        } src;
        union {
            struct rte_ipv6_addr v6;
            ip4_addr_t v4;  
        } dst;
        uint16_t len;
        union {
            uint8_t hop_limit;
            uint8_t ttl;
        };
        uint8_t proto;
});
```

Reply:
oh dear. I think I prefer the `packet_type` hacking, even though it is ugly.